clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,506 @@
+	npm.podInfo.podPairCh = make(chan uniquePodPair)
+}
+
+func (npm *networkPerformanceMeasurement) initializeInformer() error {
+	gvr, _ := meta.UnsafeGuessKindToResource(crGvk)
+	npm.resourceInterface = npm.dynamicClient.Resource(gvr).Namespace(netperfNamespace)
+	informer, err := getInformer(netperfNamespace, npm.dynamicClient, gvr)
+	if err != nil {
+		return fmt.Errorf("error getting informer:%s", err)
+	}
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(oldObj interface{}, newObj interface{}) {
+			npm.handleUpdateNetworkTestEvents(newObj)
+		},
+	})
+	npm.stopCh = make(chan struct{})
+	go informer.Run(npm.stopCh)
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) prepareCluster() error {
+	if err := client.CreateNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		return fmt.Errorf("error while creating namespace: %v", err)
+	}
+
+	// Create clusterRoleBinding
+	if err := npm.framework.ApplyTemplatedManifests(clusterRoleBindingFilePath, npm.templateMapping); err != nil {
+		return fmt.Errorf("error while creating clusterRoleBinding: %v", err)
+	}
+
+	// Create CRD
+	if err := npm.framework.ApplyTemplatedManifests(crdManifestFilePath, npm.templateMapping); err != nil {
+		return fmt.Errorf("error while creating CRD: %v", err)
+	}
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) cleanupCluster() {
+	// Delete CRD
+	if err := npm.framework.DeleteObject(crdGvk, "", customResourceDefinitionName); err != nil {
+		klog.Errorf("error while deleting CRD: %v", err)
+	}
+	// Delete clusterRoleBinding
+	if err := npm.framework.DeleteObject(rbacGvk, "", rbacName); err != nil {
+		klog.Errorf("error while deleting clusterRoleBinding: %v", err)
+	}
+	if err := client.DeleteNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		klog.Errorf("error while deleting %s namespace: %v", netperfNamespace, err)
+	}
+	if err := client.WaitForDeleteNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		klog.Errorf("error while waiting for %s namespace to be deleted: %v", netperfNamespace, err)
+	}
+}
+
+func (npm *networkPerformanceMeasurement) createAndWaitForWorkerPods() error {
+	// Create worker pods
+	npm.templateMapping = map[string]interface{}{"Replicas": npm.numberOfClients + npm.numberOfServers}
+	if err := npm.framework.ApplyTemplatedManifests(workerPodDeploymentManifestFilePath, npm.templateMapping); err != nil {
+		return fmt.Errorf("error while creating worker pods :%v ", err)
+	}
+
+	// Wait for all worker pods to be ready
+	weightedPodReadyTimeout := npm.getWeightedPodReadyTimeout()
+	stopCh := make(chan struct{})
+	time.AfterFunc(weightedPodReadyTimeout, func() {
+		close(stopCh)
+	})
+	options := &measurementutil.WaitForPodOptions{
+		Selector:            &measurementutil.ObjectSelector{Namespace: netperfNamespace},
+		DesiredPodCount:     npm.numberOfClients + npm.numberOfServers,
+		CallerName:          networkPerformanceMetricsName,
+		WaitForPodsInterval: podReadyPollInterval,
```

Comment:
let's inline that 2s here (not define podReadyPollInterval - it's used here only)

Reply:
Done